### PR TITLE
Extend interpolation test suite and fix bug

### PR DESCRIPTION
**Pull request by BjoernLudwigPTB**
This introduces the fix for an important corner case of parameter selection, in which we reveive a `NotImplementedError`, which is not true. The according call would be:

```python
interp1d_unc(t_new, t=t, y=y, uy=uy, returnC=True)
```

The cause of this is treating `bounds_error`'s default value `None` like an intentional `False`. In this PR we [correct the according Hypothesis strategy to cover these cases](8118f8b3bd6fe783eb5973f5b8789f1840943310) and [fail as expected](https://app.circleci.com/pipelines/github/PTB-PSt1/PyDynamic/844/workflows/b34a1aa3-1cc9-4fd1-8942-578ba297a16e) with the original implementation. Then we [adapt the implementation and fix the bug](https://github.com/PTB-PSt1/PyDynamic/pull/168/files/8118f8b3bd6fe783eb5973f5b8789f1840943310..ae382167a3ebc64e9a9a2b685d3e54c4efd4ed60) so [the tests pass](https://app.circleci.com/pipelines/github/PTB-PSt1/PyDynamic/847/workflows/b21c4c57-14ef-46af-805c-1e4c9b75d7b2) again.

### CODE REVIEW

File: PyDynamic/uncertainty/interpolate.py [hunk 1]
```diff
@@ -169,8 +169,13 @@ def interp1d_unc(
 
         if fill_unc == "extrapolate":
             fill_unc = uy[0], uy[-1]
-        elif returnC:
-            # Once we deal with this, we will probably introduce another input parameter
+        elif bounds_error is not None and returnC:
```
Comment (mgrub) [inline PyDynamic/uncertainty/interpolate.py:172]: If we use an `elif` here, this case gets only checked if `fill_unc != "extrapolate"`. Is this the intended behavior?

**Review by mgrub — COMMENTED** [review]
Thanks for fixing this bug and extending the relevant comment in the source.

**Review by mgrub — APPROVED** [review]

File: PyDynamic/uncertainty/interpolate.py (hunk 1 above)
Comment (BjoernLudwigPTB) [reply PyDynamic/uncertainty/interpolate.py:172]: TL;DR
Yes, it is.

Check to additionally added comment in the following lines for that.
